### DISCUSSION TITLE
mysql 连接数过多

### DISCUSSION THREAD

**Discussion by huangshanhui** [juicedata/juicefs#2918] [Q&A] [answered]
看日志时发现 mysql 连接数量过多导致 pod 无法挂载 pvc，进入 mysql 后基本都是 Sleep 的连接（超过 2000），单个 mount pod 内的连接数最多有上百，某些 mount pod 的连接似乎没有业务使用也未释放（发现有一个节点的 mount pod 只有一个卡在 Init 状态的应用 pod 在该节点挂载了对应 pv，但是 mount pod 内的连接数仍然有 60 多），想请问下有什么方法可以排查下原因或者解决？

另外部分 mount pod 一直报错：update session: no session found matching sid 这个是否有影响

**Comment by zhijian-pro** [comment]
csi 版本多少？

**Reply by huangshanhui** [reply]
v0.15.0

**Reply by zhijian-pro** [reply]
这个是 mysql 连接相关的配置
```
        engine.DB().SetMaxIdleConns(runtime.NumCPU() * 2)
	engine.DB().SetConnMaxIdleTime(time.Minute * 5)
```
理论上，CSI 并不会造成 mysql 的连接数过多，所以可以尝试在本地复现这个问题。
1. 本地挂载， 进入 mysql 观察当前的连接数，记录下juicefs 使用的的 sessionid
2. 在挂载点使用  juicefs bench -p  模拟并发，然后观察新增的mysql 连接，记录下新增的 sessionid
3. 等待bench 命令跑完后，会有大量连接被空闲，所以观察第二步新增的这些 session 的空闲时间是否会超过五分钟，理论上 5 分钟后会被关闭。

**Reply by huangshanhui** [reply]
好的，我在本地尝试复现一下

**Comment by zhijian-pro** [comment, marked as answer]
这个问题已经确定了，是 go1.17 标准库的 [bug](https://github.com/golang/go/commit/74f99d0933d5c201fc17d90ab612cd1a9c7d425f)，使用 go 1.18 以上的go 版本编译是没有问题的。你可以先手动编译解决这个问题。后续我们会在release 分支上修复。
这个问题目前已经修复了 #2938
